flowey/flowey_lib_common/src/publish_test_results.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Publish test results.
+//!
+//! - On ADO, this will hook into the backend's native JUnit handling.
+//! - On Github, this will publish artifacts containing the raw JUnit XML file
+//!   and any optional attachments.
+//! - When running locally, this will optionally copy the XML files and any
+//!   attachments to the provided artifact directory.
+
+use crate::_util::copy_dir_all;
+use flowey::node::prelude::*;
+use std::collections::BTreeMap;
+
+flowey_request! {
+    pub enum Request {
+        /// Register a XML file to be published
+        Register {
+            /// Path to a junit.xml file
+            ///
+            /// HACK: this is an optional since `flowey` doesn't (yet?) have any way
+            /// to perform conditional-requests, and there are instances where nodes
+            /// will only conditionally output JUnit XML.
+            ///
+            /// To keep making forward progress, I've tweaked this node to accept an
+            /// optional... but this ain't great.
+            junit_xml: ReadVar<Option<PathBuf>>,
+            /// Brief string used when publishing the test.
+            test_label: String,
+            /// Additional attachments for platforms without JUnit integration (not used on ADO)
+            attachments: Option<BTreeMap<String, ReadVar<PathBuf>>>,
```

Comment:
no need to wrap the btreemap in an option. just pass an empty btreemap if there aren't any attachements